config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/beans/PolicyType.java

```diff
@@ -15,27 +19,38 @@
 import static org.apache.commons.lang3.StringUtils.isEmpty;
 
 public enum PolicyType {
-    INCLUDE("Include"),
-    SERVICE_OPERATION("Service Operation"),
-    GLOBAL("Global"),
-    INTERNAL("Internal",
+    INCLUDE("Include", Policy.class),
+    SERVICE_OPERATION("Service Operation", Policy.class),
+    GLOBAL("Global", GlobalPolicy.class),
+    INTERNAL("Internal", AuditPolicy.class,
             "audit-lookup",
             "audit-sink",
             "audit-message-filter",
             "audit-viewer");
 
     private String type;
     private List<String> tags;
+    private Class<? extends Policy> policyClass;
 
-    PolicyType(String type, String... tags) {
+    PolicyType(String type, Class<? extends Policy> policyClass, String... tags) {
         this.type = type;
+        this.policyClass = policyClass;
         this.tags = new ArrayList<>(asList(firstNonNull(tags, new String[0])));
     }
 
     public String getType() {
         return type;
     }
 
+    public Policy createPolicyObject() {
```

Comment:
Should we return the policy builder here instead?

Reply:
if I change to the builder I will have to change the loader to set things in the builder, it's an approach but not really needed for now... what do you think?

Reply:
I think it'd be nice to have since the loader does a lot of setting, which would be perfect for a builder. But I don't know if there's a neat way to use the builder through reflections

Reply:
I tried here but the refactoring will be too much for this PR, I will fix this in the future.